examples/door-lock-app/linux/src/LockEndpoint.cpp

```diff
@@ -287,19 +287,11 @@ DlStatus LockEndpoint::SetSchedule(uint8_t yearDayIndex, uint16_t userIndex, DlS
 
 bool LockEndpoint::setLockState(DlLockState lockState, const Optional<chip::ByteSpan> & pin, DlOperationError & err)
 {
-    if (mLockState == lockState)
-    {
-        ChipLogDetail(Zcl, "Door Lock App: door is already locked, ignoring command to set lock state to \"%s\" [endpointId=%d]",
-                      lockStateToString(lockState), mEndpointId);
-        return true;
-    }
-
     if (!pin.HasValue())
     {
-        ChipLogDetail(Zcl, "Door Lock App: PIN code is not specified, setting door lock state to \"%s\" [endpointId=%d]",
+        ChipLogDetail(Zcl, "Door Lock App: PIN code is not specified, not setting door lock state to \"%s\" [endpointId=%d]",
```

Comment:
Why this change?  If a PIN is not specified, then it's not supposed to be checked, but the operation still carried out, no?

Reply:
Yes, you are correct. If there is no PIN passed, and RequirePINforRemoteOperation is set to true, at the server level, it will return kInvalidCredential, but if RequirePINforRemoteOperation is false and No PIN is provided, then the server calls this function with no PIN. Good catch! Thank you!